--- ntp-4.2.4p2/Makefile.in.nosntp	2007-06-20 13:25:39.000000000 +0200
+++ ntp-4.2.4p2/Makefile.in	2007-08-21 14:58:49.000000000 +0200
@@ -215,7 +215,7 @@ ACLOCAL_AMFLAGS = -I m4 -I libopts/m4
 NULL = 
 SUBDIRS = $(am__append_1) scripts include ElectricFence @ARLIB_DIR@ \
 	libntp libparse ntpd ntpdate ntpdc ntpq parseutil adjtimed \
-	clockstuff kernel sntp util $(NULL)
+	clockstuff kernel util $(NULL)
 DIST_SUBDIRS = \
 	scripts		\
 	include		\
@@ -232,7 +232,6 @@ DIST_SUBDIRS = \
 	adjtimed	\
 	clockstuff	\
 	kernel		\
-	sntp		\
 	util		\
 	$(NULL)
 
--- ntp-4.2.4p2/configure.nosntp	2007-06-20 13:25:58.000000000 +0200
+++ ntp-4.2.4p2/configure	2007-08-21 14:58:34.000000000 +0200
@@ -53073,9 +53073,9 @@ _ACEOF
 
 
 
-ac_subdirs_all="$ac_subdirs_all sntp"
+ac_subdirs_all="$ac_subdirs_all"
 
-subdirs="$subdirs sntp"
+subdirs="$subdirs"
 
 
 cat >confcache <<\_ACEOF
