--- ntp-4.1.1c-rc2/ntpdate/ntpdate.c.authkey	2003-07-01 11:12:38.000000000 +0200
+++ ntp-4.1.1c-rc2/ntpdate/ntpdate.c	2003-07-01 11:33:39.000000000 +0200
@@ -611,7 +611,7 @@
 		}
 		authtrust(sys_authkey, 1);
 		if (!authistrusted(sys_authkey)) {
-			char buf[10];
+			char buf[16]; /* is 11, but better safe than sorry */
 
 			(void) sprintf(buf, "%lu", (unsigned long)sys_authkey);
 			msyslog(LOG_ERR, "authentication key %s unknown", buf);
