diff -up ntp-4.2.4p4/configure.ipv6 ntp-4.2.4p4/configure
--- ntp-4.2.4p4/configure.ipv6	2008-03-10 13:39:45.000000000 +0100
+++ ntp-4.2.4p4/configure	2008-03-10 16:15:39.000000000 +0100
@@ -52780,15 +52780,19 @@ cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 
+#define _GNU_SOURCE
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
+;
 $isc_netinetin6_hack
 $isc_netinet6in6_hack
 
+;
 int
 main ()
 {
+;
 struct in6_pktinfo xyzzy; return (0);
   ;
   return 0;
