--- ntp-4.2.4p0/ntpd/ntp_io.c.ifupdate	2007-01-30 12:02:37.000000000 +0100
+++ ntp-4.2.4p0/ntpd/ntp_io.c	2007-05-17 16:32:38.000000000 +0200
@@ -1243,7 +1243,9 @@
 		 * with the appropriate socket (our name space is
 		 * (ip-address) - NOT (interface name, ip-address))
 		 */
-		iface = getinterface(&interface.sin);
+		iface = find_addr_in_list(&interface.sin);
+		if (iface && iface->flags & INT_WILDCARD)
+			iface = NULL;
 		
 		if (iface && refresh_interface(iface)) 
 		{
