diff -up ntp-4.2.4p4/configure.gettime ntp-4.2.4p4/configure
--- ntp-4.2.4p4/configure.gettime	2008-07-24 19:12:26.000000000 +0200
+++ ntp-4.2.4p4/configure	2008-07-24 19:12:27.000000000 +0200
@@ -43414,7 +43414,10 @@ return f != $ac_func;
 _ACEOF
 rm -f conftest.$ac_objext conftest$ac_exeext
 if { (eval echo "$as_me:$LINENO: \"$ac_link\"") >&5
+  _libs=$LIBS
+  LIBS="$LIBS -lrt"
   (eval $ac_link) 2>conftest.er1
+  LIBS=$_libs
   ac_status=$?
   grep -v '^ *+' conftest.er1 >conftest.err
   rm -f conftest.er1
